src/GraphQLParser/ParserContext.Parse.cs

```diff
@@ -326,28 +357,40 @@ private bool ParseRepeatable()
             return false; // for compiler
         }
 
-        private List<GraphQLName> ParseDirectiveLocations()
+        private GraphQLDirectiveLocations ParseDirectiveLocations()
         {
-            var locations = new List<GraphQLName>();
+            IncreaseDepth();
+
+            int start = _currentToken.Start;
+
+            var directiveLocations = NodeHelper.CreateGraphQLDirectiveLocations(_ignoreOptions);
+
+            var items = new List<DirectiveLocation>();
 
             // Directive locations may be defined with an optional leading | character
             // to aid formatting when representing a longer list of possible locations
             Skip(TokenKind.PIPE);
 
             do
             {
-                locations.Add(ParseName());
```

Comment:
Important thing here - _location_ **is not** a _GraphQLName_. It was a bug. Location is _oneOf_ term - closed set of predefined values. I'll add the same fix for _OperationType_ later in further PR.